docs/doxygen/migration/mqtt.txt

```diff
@@ -0,0 +1,127 @@
+/**
+@page mqtt_migration MQTT
+@brief How to migrate an MQTT application from v3 to 202009.00.
+
+The MQTT library has been redesigned in 202009.00 as a solely single-threaded library. New features such as persistent session support and QoS 2 publish delivery have also been added.
```

Comment:
The use of "as a solely single-threaded library" could be misleading to customers as the library being suitable only for single-threaded applications.  Instead, we can use a phrase that mentions that library has been re-designed to "have no threading synchronization dependencies to be light weight"

@dan4thewin , what are your thoughts on the wording?

Reply:
re-factored instead of re-designed.

Reply:
Rephrased to "The MQTT library has been refactored in 202009.00 to use only a single thread, and therefore no longer has any thread synchronization dependencies"